src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/test_resource.py

```diff
@@ -67,63 +67,47 @@ def body(self):
         s.cmd('group exists -n {}'.format(rg), checks=NoneCheck())
 
 
-class ResourceScenarioTest(ResourceGroupVCRTestBase):
-    def test_resource_scenario(self):
-        self.execute()
-
-    def __init__(self, test_method):
-        super(ResourceScenarioTest, self).__init__(__file__, test_method,
-                                                   resource_group='cli_test_resource_scenario')
-        self.vnet_name = 'cli-test-vnet1'
-        self.subnet_name = 'cli-test-subnet1'
+class ResourceScenarioTest(ScenarioTest):
+    @ResourceGroupPreparer(name_prefix='cli_test_rsrc_scenario')
+    def test_resource_scenario(self, resource_group):
+        vnet_name = self.create_random_name('cli-test-vnet', 30)
+        subnet_name = self.create_random_name('cli-test-subnet', 30)
+        vnet_type = 'Microsoft.Network/virtualNetworks'
 
-    def set_up(self):
-        super(ResourceScenarioTest, self).set_up()
-
-    def body(self):
-        s = self
-        rg = self.resource_group
-        vnet_count = s.cmd(
-            "resource list --query \"length([?name=='{}'])\"".format(self.vnet_name)) or 0
         self.cmd('network vnet create -g {} -n {} --subnet-name {} --tags cli-test=test'.format(
-            self.resource_group, self.vnet_name, self.subnet_name))
-        vnet_count += 1
-
-        s.cmd('resource list',
-              checks=JMESPathCheck("length([?name=='{}'])".format(self.vnet_name), vnet_count))
-        s.cmd('resource list -l centralus',
-              checks=JMESPathCheck("length([?location == 'centralus']) == length(@)", True))
-        s.cmd('resource list --resource-type Microsoft.Network/virtualNetworks',
-              checks=JMESPathCheck("length([?name=='{}'])".format(self.vnet_name), vnet_count))
-        s.cmd('resource list --name {}'.format(self.vnet_name),
-              checks=JMESPathCheck("length([?name=='{}'])".format(self.vnet_name), vnet_count))
-        s.cmd('resource list --tag cli-test',
-              checks=JMESPathCheck("length([?name=='{}'])".format(self.vnet_name), vnet_count))
-        s.cmd('resource list --tag cli-test=test',
-              checks=JMESPathCheck("length([?name=='{}'])".format(self.vnet_name), vnet_count))
+            resource_group, vnet_name, subnet_name))
+
+        self.cmd('resource list', checks=JCheck("length([?name=='{}'])".format(vnet_name), 1))
+        self.cmd('resource list -l centralus', checks=JCheck("length([?location == 'centralus']) == length(@)", True))
```

Comment:
Can we use `southcentralus` instead? The traffic to that location is lower and is recommended for automation and internal use.

Reply:
I was not aware, can def do.